test/Mono.Linker.Tests.Cases/UnreachableBlock/MultiStageRemoval.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.UnreachableBlock
+{
+	public class MultiStageRemoval
+	{
+		public static void Main()
+		{
+			TestMethod_1 ();
+			TestMethod_2 ();
+		}
+
+		[Kept]
+		[ExpectBodyModified]
+		static void TestMethod_1 ()
+		{
+			if (TestProperty_int () == 0)
```

Comment:
Once we got into more complex body editing and leveraging `[ExpectBodyModified]` I noticed it wasn't that hard to have bugs that resulted in valid IL but the modified IL was wrong.  If you were really diligent with your test setup and calling methods and should be removed or kept you could prevent most bugs from slipping by.  However, for added safety we added a series of attributes to the test framework to invoke the linked code.

We a few attributes that will load the input and output assemblies in an AppDomain and invoke methods and check their behavior.
`[CheckInvokeMatchesOriginal]`
`[CheckInvoke(<expected result>)]`
`[CheckInvokeThrows]`

And we have one other attribute that you can put on the test case type itself that will tell the test framework to run the input and output `test.exe` and compare their output.
`[CheckOutput]`

If you would like me to bring these upstream let me know.

Reply:
I think peverify is able to catch such issue so no for now